File: backend/app/schemas/domain.py
"""
Pydantic schemas for the Domain resource.

Domains are seeded reference data - no create or update
endpoints are exposed via the API. Only a Response schema
is needed for GET /domains.
"""

from uuid import UUID
from app.schemas.common import ORMBase


class DomainResponse(ORMBase):
    """Response shape for GET /domains."""

    id: UUID
    name: str
    display_name: str
    description: str | None
    uc_schema_name: str

File: frontend/src/types/domain.ts
export interface Domain {
  /** UUID v4. Backend returns it as a string. */
  id: string

  /** Human-readable domain name (e.g. "Finance"). */
  name: string

  /** Unity Catalog schema name (e.g. "finance"). */
  uc_schema_name: string

  description: string | null
}